sysrib/server.go

```diff
@@ -88,18 +95,87 @@ func (d *Dataplane) ProgramRoute(r *ResolvedRoute) error {
 // NewServer instantiates server to handle client queries.
 //
 // If dp is nil, then a connection attempt is made.
-func NewServer(dp dataplaneAPI) (*Server, error) {
+func NewServer(dp dataplaneAPI, gnmiServerAddr, target string) (*Server, error) {
 	rib, err := NewSysRIB(nil)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Server{
+	s := &Server{
 		rib:            rib,
 		interfaces:     map[Interface]bool{},
 		resolvedRoutes: map[RouteKey]*ResolvedRoute{},
 		dataplane:      dp,
-	}, nil
+	}
+	if err := s.monitorConnectedIntfs(gnmiServerAddr, target); err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+// monitorConnectedIntfs starts a gothread to check for connected prefixes from
+// connected interfaces and adds them to the sysrib. It returns an error if
+// there is an error before monitoring can begin.
+//
+// - gnmiServerAddr is the address of the central gNMI datastore.
+// - target is the name of the gNMI target.
+func (s *Server) monitorConnectedIntfs(gnmiServerAddr, target string) error {
+	var opts []grpc.DialOption
+	opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
this won't work if tls is enabled

Reply:
I'll also fix this in the next PR as part of refactoring.